app/models/slide.rb

```diff
@@ -99,4 +96,8 @@ def update_file
       SlideConvertJob.perform_later(id)
     end
   end
+
+  def should_generate_new_friendly_id?
+    slug.blank? || title_changed?
+  end
 end
```

Comment:
FYI, it better to define a concern such as "FriendlyUrlSlug", to DRY these changes